vortex/src/main/java/org/apache/iceberg/vortex/VortexIterable.java

```diff
@@ -58,8 +66,7 @@ public class VortexIterable<T> extends CloseableGroup implements CloseableIterab
   @Override
   public CloseableIterator<T> iterator() {
     // TODO(aduffy): this HadoopInputFile assumption is maybe silly.
-    String path = ((HadoopInputFile) inputFile).getPath().toUri().getPath();
-    NativeFile vortexFile = NativeFile.open(path);
+    File vortexFile = newVortexFile(inputFile);
```

Comment:
This is ok for now, but for the long run we need to use the standard InputFile interface:
- long getLength();
- SeekableInputStream newStream();
- String location();
- boolean exists();

In this case, we don't have to bother with all of the access related stuff (handled by the FileIO/InputFile). We just have to rely on the `SeekableInputStream`.
Docs:
- https://iceberg.apache.org/fileio/ - just a short intro
- https://www.tabular.io/blog/iceberg-fileio-cloud-native-tables/ - longer intro
- https://github.com/apache/iceberg/pull/8303 - ADLSv2 PR which added support for it
- Or you can use HadoopFileIO which provides a way to access any hadoop compatible FS

Reply:
Yea I'm sure we can get there, I think it is going to take a fair bit of work on the Rust side to make that possible. Right now much like Comet with their NativeBatchReader, we implement scanning directly in Rust and just expose that as a ColumnarBatch to consumers. Our Rust interface for reading looks a lot like SeekableInputStream though so there is probably a reasonable way to bridge those worlds